horovod/ray/elastic.py

```diff
@@ -0,0 +1,300 @@
+import logging
+import time
+import os
+import math
+from typing import Callable, List, Any, Dict, Optional
+
+from horovod.runner.common.util import timeout, secret
+
+from horovod.runner.http.http_server import RendezvousServer
+from horovod.runner.util import network
+from horovod.runner.gloo_run import (create_slot_env_vars, create_run_env_vars,
+                                     _get_min_start_hosts)
+from horovod.runner.driver import driver_service
+from horovod.runner.elastic.settings import ElasticSettings
+from horovod.runner.elastic.rendezvous import create_rendezvous_handler
+from horovod.runner.elastic.discovery import HostDiscovery
+from horovod.runner.elastic.driver import ElasticDriver
+
+import ray
+import ray.exceptions
+from horovod.ray.runner import BaseHorovodWorker
+
+logger = logging.getLogger(__name__)
+
+if hasattr(ray.exceptions, "GetTimeoutError"):
+    GetTimeoutError = ray.exceptions.GetTimeoutError
+elif hasattr(ray.exceptions, "RayTimeoutError"):
+    GetTimeoutError = ray.exceptions.RayTimeoutError
+else:
+    raise ImportError("Unable to find Ray Timeout Error class "
```

Comment:
Is there a min version of Ray we should set in `setup.py`?

Reply:
well this specific handling allows compat between >1.0 and <1.0; the overall integration should be compatible with releases since 0.8.0 (released a long while ago)